server/src/models/swing.rs

```diff
@@ -199,80 +197,63 @@ impl Swing {
 
 /// Compute the minimum value for a time series segment whose values are
 /// represented by a model of type Swing.
-pub fn min(
-    _gid: TimeSeriesId,
-    start_time: Timestamp,
-    end_time: Timestamp,
-    _sampling_interval: i32,
-    model: &[u8],
-    _gaps: &[u8],
-) -> f32 {
+pub fn min(start_time: Timestamp, end_time: Timestamp, model: &[u8]) -> Value {
     let (slope, intercept) = decode_model(model);
     if slope == 0.0 {
-        intercept as f32
+        intercept as Value
     } else if slope > 0.0 {
-        (slope * start_time as f64 + intercept) as f32
+        (slope * start_time as f64 + intercept) as Value
     } else {
-        (slope * end_time as f64 + intercept) as f32
+        (slope * end_time as f64 + intercept) as Value
     }
 }
 
 /// Compute the maximum value for a time series segment whose values are
 /// represented by a model of type Swing.
-pub fn max(
-    _gid: TimeSeriesId,
-    start_time: Timestamp,
-    end_time: Timestamp,
-    _sampling_interval: i32,
-    model: &[u8],
-    _gaps: &[u8],
-) -> f32 {
+pub fn max(start_time: Timestamp, end_time: Timestamp, model: &[u8]) -> Value {
     let (slope, intercept) = decode_model(model);
     if slope == 0.0 {
-        intercept as f32
+        intercept as Value
     } else if slope < 0.0 {
-        (slope * start_time as f64 + intercept) as f32
+        (slope * start_time as f64 + intercept) as Value
     } else {
-        (slope * end_time as f64 + intercept) as f32
+        (slope * end_time as f64 + intercept) as Value
     }
 }
 
 /// Compute the sum of the values for a time series segment whose values are
 /// represented by a model of type Swing.
 pub fn sum(
-    _gid: TimeSeriesId,
     start_time: Timestamp,
     end_time: Timestamp,
     sampling_interval: i32,
     model: &[u8],
-    _gaps: &[u8],
-) -> f32 {
+) -> Value {
     let (slope, intercept) = decode_model(model);
     let first = slope * start_time as f64 + intercept;
     let last = slope * end_time as f64 + intercept;
     let average = (first + last) / 2.0;
-    let length = ((end_time - start_time) / sampling_interval as i64) + 1;
-    (average * length as f64) as f32
+    let length = models::length(start_time, end_time, sampling_interval);
+    (average * length as f64) as Value
 }
 
 /// Reconstruct the data points for a time series segment whose values are
 /// represented by a model of type Swing. Each data point is split into its
 /// three components and appended to `tids`, `timestamps`, and `values`.
 pub fn grid(
-    gid: TimeSeriesId,
+    tid: TimeSeriesId,
```

Comment:
Since this has been changed to `time_series_id` in `mod.rs` it should maybe also be changed in the individual model files.

Reply:
Fixed.